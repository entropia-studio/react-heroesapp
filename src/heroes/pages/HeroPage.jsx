import React from 'react';

export const HeroPage = () => {
  return <h1>HeroePage</h1>;
};
