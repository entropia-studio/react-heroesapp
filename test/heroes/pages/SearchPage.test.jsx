import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes/pages/SearchPage';

describe('SearchPage', () => {
  test('debe de mostrarse correctamente con los valores por defecto', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrarse Batman y el input con el valor del query string', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=Batman']}>
        <SearchPage />
      </MemoryRouter>
    );
    const inputValue = screen.getByRole('textbox');
    expect(inputValue.value).toBe('Batman');
    const imgElement = screen.getByRole('img');
    expect(imgElement.src).toContain('/assets/heroes/dc-batman.jpg');
    const searchDiv = screen.getByLabelText('Search a hero');
    expect(searchDiv.style).toHaveProperty('display', 'none');
  });
});
