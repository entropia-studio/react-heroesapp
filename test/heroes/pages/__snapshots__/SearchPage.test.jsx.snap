// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage debe de mostrarse correctamente con los valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary btn-block mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="Search a hero"
        class="alert alert-primary animate_animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate_animated animate__fadeIn"
        style="display: none;"
      >
        This hero 
        <b />
         doesn't exist
      </div>
    </div>
  </div>
</div>
`;
